engine/types/type42/linking_key_test.go

```diff
@@ -0,0 +1,86 @@
+package type42
+
+import (
+	"encoding/hex"
+	"testing"
+
+	"github.com/libsv/go-bk/bec"
+	assert "github.com/stretchr/testify/require"
+)
+
+func TestDeriveLinkedKey(t *testing.T) {
+	sourcePubKeyHex := "027c1404c3ecb034053e6dd90bc68f7933284559c7d0763367584195a8796d9b0e"
+	sourcePubKeyBytes, err := hex.DecodeString(sourcePubKeyHex)
+	assert.NoError(t, err)
+	sourcePubKey, err := bec.ParsePubKey(sourcePubKeyBytes, bec.S256())
+	assert.NoError(t, err)
+
+	linkPubKeyHex := "03a34e456deecb6e6e9237e63e5b7d045d1d2a456eb6be43de1ec4e9ac9a07b50d"
+	linkPubKeyBytes, err := hex.DecodeString(linkPubKeyHex)
+	assert.NoError(t, err)
+	linkPubKey, err := bec.ParsePubKey(linkPubKeyBytes, bec.S256())
+	assert.NoError(t, err)
+
+	validHMAC, err := calculateHMAC(sourcePubKey.SerialiseCompressed(), "valid-invoice")
+	assert.NoError(t, err)
+	validDerivedKey, err := calculateLinkedPublicKey(validHMAC, linkPubKey)
+	assert.NoError(t, err)
```

Comment:
Still I don't like this approach to tests.
IMHO, we should have list of cases, where we have sender and recipient public keys, invoice number and expected derivedPublicKey.

And this is list of test cases that I generated from code from ts-sdk adjusted to our solution for derivations.
```
[
  {
    senderPublicKey: '033f9160df035156f1c48e75eae99914fa1a1546bec19781e8eddb900200bff9d1',
    recipientPublicKey: '02133b035cda4ba15f93b5fdde11c1f73eb9f1a79b60c6caa1c78e1c4c64ed72ce',
    invoiceNumber: 'f3WCaUmnN9U=',
    derivedPublicKey: '02d4049747553b4b956a419a7e1ddef418d57f317de8cc7d024d05c75f29309f26',
  },
  {
    senderPublicKey: '027775fa43959548497eb510541ac34b01d5ee9ea768de74244a4a25f7b60fae8d',
    recipientPublicKey: '02dfcbe35d95b55b5f3168ea8f12717e266ceddf88d04d2ff741272dfb0e542c2a',
    invoiceNumber: '2Ska++APzEc=',
    derivedPublicKey: '02679645bc44a771154f66eb52ca93507c3e777997a774e87611a7b17d30d748ee',
  },
  {
    senderPublicKey: '0338d2e0d12ba645578b0955026ee7554889ae4c530bd7a3b6f688233d763e169f',
    recipientPublicKey: '023c1db4cce57a44c95e05f0ed499085a9ce8ac83bf80dd4c7b4658a9d6c4a122e',
    invoiceNumber: 'cN/yQ7+k7pg=',
    derivedPublicKey: '03c07884a8e8f02bb1cfb276edddfa8dade7654d8cb744e0ff9e4b0c5dffa63ae0',
  },
  {
    senderPublicKey: '02830212a32a47e68b98d477000bde08cb916f4d44ef49d47ccd4918d9aaabe9c8',
    recipientPublicKey: '0363ec92e374974e9c904875ff8daff50bfa22d8ec0a39891a31e5e760aca5e9cf',
    invoiceNumber: 'm2/QAsmwaA4=',
    derivedPublicKey: '024d369f8440e91fdd58577bd6bc471d868fdfc58e605eb2ec5a2362df589d43cd',
  },
  {
    senderPublicKey: '03f20a7e71c4b276753969e8b7e8b67e2dbafc3958d66ecba98dedc60a6615336d',
    recipientPublicKey: '02cef40bd6826d1b1960bc94094c0c0a19547b291c33b7a07380cbdc4580f3678b',
    invoiceNumber: 'jgpUIjWFlVQ=',
    derivedPublicKey: '03aa3ea0de12f44642a51593b2f8a0e0ff9813cdb4ab39ab8039654dce2ae5bdb3',
  },
  {
    senderPublicKey: '03f20a7e71c4b276753969e8b7e8b67e2dbafc3958d66ecba98dedc60a6615336d',
    recipientPublicKey: '0363ec92e374974e9c904875ff8daff50bfa22d8ec0a39891a31e5e760aca5e9cf',
    invoiceNumber: 'jgpUIjWFlVQ=',
    derivedPublicKey: '032ac4dccb237d777361a7fade10fc2d21b642df9b0e4634d3e27fdce131526e4f',
  },
  {
    senderPublicKey: '02830212a32a47e68b98d477000bde08cb916f4d44ef49d47ccd4918d9aaabe9c8',
    recipientPublicKey: '02cef40bd6826d1b1960bc94094c0c0a19547b291c33b7a07380cbdc4580f3678b',
    invoiceNumber: 'jgpUIjWFlVQ=',
    derivedPublicKey: '02caa04e2020a0642e5283c978257d2135c00882c049adcd71a71dffe9d7208a39',
  },
  {
    senderPublicKey: '02830212a32a47e68b98d477000bde08cb916f4d44ef49d47ccd4918d9aaabe9c8',
    recipientPublicKey: '02cef40bd6826d1b1960bc94094c0c0a19547b291c33b7a07380cbdc4580f3678b',
    invoiceNumber: 'm2/QAsmwaA4=',
    derivedPublicKey: '030467472149ac58d9d04e4182b03af99593a7af312623c4be7d96f2fde08f6421',
  },
]
```
Public keys are serialised compressed. I hope I did it correctly.